examples/audio/mic/source/main.c

```diff
@@ -36,45 +47,42 @@ int main()
 		if (kDown & KEY_START)
 			break; // break in order to return to hbmenu
 
-		if(audio_initialized)
+		if(initialized)
 		{
-			framebuf = gfxGetFramebuffer(GFX_TOP, GFX_LEFT, NULL, NULL);
-
 			if(kDown & KEY_A)
 			{
 				audiobuf_pos = 0;
-				printf("Stopping audio playback\n");
-				CSND_SetPlayState(0x8, 0);//Stop audio playback.
-				CSND_UpdateInfo(0);
+				micbuf_pos = 0;
 
-				MIC_SetRecording(1);
+				printf("Stopping audio playback...\n");
+				CSND_SetPlayState(0x8, 0);
+				if(R_FAILED(CSND_UpdateInfo(0))) printf("Failed to stop audio playback.\n");
 
-				memset(framebuf, 0x20, 0x46500);
-				printf("Now recording\n");
+				printf("Starting sampling...\n");
+				if(R_SUCCEEDED(MICU_SetPower(true)) && R_SUCCEEDED(MICU_SetGain(0x40)) && R_SUCCEEDED(MICU_StartSampling(ENCODING_PCM16_SIGNED, SAMPLE_RATE_16360, 0, micbuf_datasize, true))) printf("Now recording.\n");
+				else printf("Failed to start sampling.\n");
 			}
 
 			if((hidKeysHeld() & KEY_A) && audiobuf_pos < audiobuf_size)
 			{
-				audiobuf_pos+= MIC_ReadAudioData(&audiobuf[audiobuf_pos], audiobuf_size-audiobuf_pos, 1);
-				if(audiobuf_pos > audiobuf_size)audiobuf_pos = audiobuf_size;
-
-				memset(framebuf, 0x60, 0x46500);
+				u32 micbuf_readpos = micbuf_pos;
+				micbuf_pos = micGetLastSampleOffset();
+				while(audiobuf_pos < audiobuf_size && micbuf_readpos != micbuf_pos)
+				{
```

Comment:
Only thing is, it can loop around to the beginning of the mic buffer. It isn't always linear.